source/develop/release-management/features/network/openvswitch/native-openvswitch.html.md

```diff
@@ -0,0 +1,200 @@
+Handling the rollback as close as possible to the setupNetworks api implementation seems more reasonable, moving this rollback handler to the network api (currently in supervdsm context) is to be considered.
+The OVS rollback is to be integrated into the existing rollback functionality without heavily affecting existing beahaviours:
+
+- OVS configurators should trigger a rollback which is to be handled at the api level (as described above).
+- The ifcfg configurator should keep the existing logic, but in addition it should inform the api level of the rollback which it handled.
+Moving the handling of the rollback from the ifcfg configurator to the api level should be investigated, simplifying rollback logic.
+- The api level should combine both ifcfg and OVS results in terms of the rollback, and issue relevant commands.
+  - When upgrading from legacy to OVS, there is a need to tear down bridges and configure OVS in one transaction and during one rollback.
+
+### VDSM API
+Minimal changes in current API verbs are required:
+
+- setupNetworks: Per network, add a switch_type key.
+  - Currently it can have two values: ‘legacy’ and ‘ovs’.
+  - If not explicitly set, assume ‘legacy’. (for compatability with Engine 3.6)
+- oVirt currently differentiates between VM (bridged) networks and non-VM networks due to historical performance benefit of non-bridged networks for non-VM payloads (such as storage and migration). This differentiation is moot with OVS, but in order to keep Engine oblivious to the change, VDSM should lie to it, and report bridged=False for networks that requested it. Dropping the differentiation from Engine should happen in a later stage.
+
+### Upgrade from native to OVS networking
+Transition between switch types must be supported for the host as a whole. Nevertheless, if the user has defined his own networking configuration which is not controlled by VDSM, it should coexist with VDSM controlled networks.
+
+Mixing between native and OVS controlled networks is not initially supported, therefore, when a network is marked with a different type from the existing, a validation check should make sure that all other networks are marked for the same network type.
+
+The following sequence describes the transition steps @Engine:
+
+- Vdsm on hosts is upgraded to an OVS-supporting version (4.0). No change takes place to preexisting network configuration, which would still be bridge-based.
+- When all active hosts in the cluster have been upgraded, the user should upgrade the cluster to version 4.0 where OVS is supported.
+- The default switch type for the cluster should be set by the user to OVS (for new hosts).
+  - New created 4.0 clusters should have OVS as the default switch.
+- Update individual hosts to transition to OVS: 
+(TBD: one or all of the options below should be chosen)
```

Comment:
We should not deliberate about this any longer. Let us perform the upgrade when no VMs are running on the host (first option). Keep the second option as "we can further improve the user experience by a second approach:"